executor/include/execution/trait/is_instance_of_base.hpp

```diff
@@ -0,0 +1,26 @@
+//
+// Created by Shrijit Singh on 14/06/20.
+//
+
+#pragma once
+
+#include <experimental/type_traits>
+#include <type_traits>
+
+namespace execution {
+
+template <template <typename...> class Type, typename Executor>
+struct is_instance_of_base : std::false_type {};
+
+template <template <typename...> class Type, typename... Args>
+struct is_instance_of_base<Type, Type<Args...>> : std::true_type {};
+
+template <template <typename...> class Type, typename Executor>
+constexpr bool is_instance_of_base_v =
+    is_instance_of_base<Type, Executor>::value;
+
+template <template <typename...> class Type, typename Executor>
+using instance_of_base =
```

Comment:
This needs to be rename to something more expressive e.g. `enable_as_int_if_instance_of_base`. Otherwise it looks like magic whenever it is used.

Reply:
The idea behind it was to be magically :wink: so that It just works since this is going to be used everywhere.
Giving a long name would really something would just make template errors longer and add additionally increase the length of code in every executor function.

Reply:
I don't agree with this trade-off. Please come up with a more expressive name. I'm ok with `enable_if_executor_t` or something along those lines, but not with the current proposal.

Reply:
Got it. Will come up with something which explains the functionality better.